warp10/src/main/java/io/warp10/script/functions/FETCH.java

```diff
@@ -514,6 +518,10 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
                 throw new WarpScriptException(getName() + " exceeded limit of " + fetchLimit + " datapoints, current count is " + fetched.get());
               }
 
+              if (stack.aborted()) {
+                throw new WarpScriptException(getName() + " execution aborted.");
+              }
+              
```

Comment:
I think it should be inside the `while` several line above, not after. It could be added after line 511 for instance.